assets/scss/components/_form.scss

```diff
@@ -21,3 +27,236 @@
         }
     }
 }
+
+.c-form__text-input,
+.c-form__password-input,
+.c-form__textarea-input {
```

Comment:
Just one more small suggestion. I'd consider using `.c-form__input .c-form__input--text` instead. Then you could extract shared styling into `.c-form__input` class and keep the code DRY :)